mozjs/js/rust/tests/magic_dom_impl.rs

```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#[macro_use] extern crate magic_codegen;
+extern crate js;
+
+use js::jsapi;
+use js::magic::{MagicSlot, SlotIndex};
+use js::rust::{GCMethods, RootKind};
```

Comment:
Are these imports depended upon by the generated code? If so, we should make the generated code use fully qualified names and/or make its own imports.

As a consumer of the macro, I imagine I'll forget to include these imports and then get really confusing error messages. Better to prevent that situation from happening altogether.